src/services/item/plugins/embeddedLink/index.ts

```diff
@@ -46,11 +46,11 @@ const plugin: FastifyPluginAsync<GraaspEmbeddedLinkItemOptions> = async (fastify
     repositories: Repositories,
     { item }: { item: Partial<Item> },
   ) => {
-    const { embeddedLink } = (item?.extra as EmbeddedLinkItemExtra) ?? {};
-
-    if (item.type !== ItemType.LINK || !embeddedLink) {
+    // if the extra is undefined or it does not contain the embedded link extra key, exit
+    if (!item.extra || !(ItemType.LINK in item.extra)) {
       return;
     }
+    const { embeddedLink } = item.extra;
```

Comment:
Does this mean that this actually type-checks without the need for the typeguards? Good news if that's the case, it would mean the type guards are not needed.

Reply:
It works here because we have 
```ts
// if the extra is undefined or it does not contain the embedded link extra key, exit
if (!item.extra || !(ItemType.LINK in item.extra)) {
  return;
}
```
above